source/dofs/dof_tools_constraints.cc

```diff
@@ -310,10 +311,9 @@ namespace DoFTools
                                            const FullMatrix<double> &face_interpolation_matrix,
                                            std::shared_ptr<std::vector<bool> > &master_dof_mask)
       {
-        if (master_dof_mask == std::shared_ptr<std::vector<bool> >())
+        if (master_dof_mask == nullptr)
           {
-            master_dof_mask = std::shared_ptr<std::vector<bool> >
-                              (new std::vector<bool> (fe1.dofs_per_face));
+            master_dof_mask.reset (new std::vector<bool> (fe1.dofs_per_face));
```

Comment:
is there a reason you opted for ``.reset(new T)`` over ``= make_shared<T>()``?

Reply:
Yes -- what I really wanted to do was replace all of these `std::shared_ptr` by `std::unique_ptr` but that doesn't work for other reasons (github issue to come soon). And I didn't want to change the code again later from `make_shared` to `make_unique`, in particular because we need to use `std_cxx14::make_unique` because `make_unique` doesn't exist in C++11.

But the long answer doesn't mean that I wouldn't change it if you insisted on it.

Reply:
I would clearly prefer using `std::make_shared`. The performance doesn't probably matter but having `new` somewhere always looks suspicious.

Reply:
No problem -- I switched this to use `make_shared`.